### PR TITLE
Add support for `ALTER TABLE <table> MODIFY COLUMN <col> <type> UNIQUE`

### PR DESCRIPTION
**Pull request by nicktobey**
This is a MySql syntax that is effectively syntactic sugar for 

```sql
ALTER TABLE <table> MODIFY COLUMN <col> <type>;
ALTER TABLE <table> ADD UNIQUE INDEX `<col>` (col);
```

### CODE REVIEW

**Comment by nicktobey** [comment]
Note that if the column already has an index with the same name as the column, this will fail with the error 
```
`col2` already exists as an index for this table
```

This is different behavior than MySQL, which will create another index with a name like `<column>_2`.

Mimicking MySQL here is doable but would require additional changes to the analyzer/executor, not just the parser, because at the parsing stage we haven't resolved the table yet and don't know what indexes it has.

Returning an error when the user is doing something that they probably aren't intending to do (adding multiple indexes to a column) that still has a workaround (calling `ADD UNIQUE INDEX` manually) seems acceptable to me.

**Comment by nicktobey** [comment]
Oh, also this fixes https://github.com/dolthub/dolt/issues/5028

(Does this work across repos? Let's find out!)

File: sql/parse/parse.go [hunk 1]
```diff
@@ -2034,8 +2080,7 @@ func tableNameToUnresolvedTableAsOf(tableName sqlparser.TableName, asOf sql.Expr
 	return plan.NewUnresolvedTableAsOf(tableName.Name.String(), tableName.Qualifier.String(), asOf)
 }
 
-func convertAlterIndex(ctx *sql.Context, ddl *sqlparser.DDL) (sql.Node, error) {
-	table := tableNameToUnresolvedTable(ddl.Table)
```
Comment (max-hoffman) [inline sql/parse/parse.go:2038]: i have a PR that rewrites this file, I'd appreciate undoing unnecessary changes that might amplify merge conflicts

File: enginetest/queries/script_queries.go [hunk 2]
```diff
@@ -1326,6 +1326,26 @@ var ScriptTests = []ScriptTest{
...
 	},
+	{
+		Name: "ALTER TABLE MODIFY column making UNIQUE",
+		SetUpScript: []string{
+			"CREATE table test (pk int primary key, uk int)",
+			"ALTER TABLE `test` MODIFY column uk int unique",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "show create table test",
```
Comment (max-hoffman) [inline enginetest/queries/script_queries.go:1337]: Again this might sound a bit orthogonal out of context, but plan tests can sometimes be harder to maintain than correctness tests. A correctness test that verifies uniqueness behavior (eg insert same row twice, check for error on second) might have a longer shelf-life because it's probably rare that we change the error behavior. A show-create might be changed less frequently than an `EXPLAIN plan`, but you could imagine how a slight tweak to how we print `SHOW CREATE` might be painful to fixup.

**Review by max-hoffman — APPROVED** [review]
LGTM, some small comments

File: enginetest/queries/script_queries.go (hunk 2 above)
Comment (nicktobey) [reply enginetest/queries/script_queries.go]: Fixed.

File: sql/parse/parse.go (hunk 1 above)
Comment (nicktobey) [reply sql/parse/parse.go]: Whoops, fixed.
